src/resolver.js

```diff
@@ -68,64 +68,68 @@ class Resolver {
          */
         const records = addresses.flat();
-        const dnslinks = records.filter((record) => dnslinkRegExp.test(record));
-
-        // If there are no dnslink records configured with our namespaces, we raise an error.
-        if (dnslinks.length === 0) {
-          return reject(
-            new NoSkynetDNSLinksFoundError(
-              `TXT records for ${lookup} found but none of them contained valid skynet dnslink - ${hint}`
-            )
-          );
-        }
+        const dnslinkSkylinks = records.filter((record) => dnslinkRegExp.test(record));
+        const dnslinkSponsors = records.filter((record) => sponsorRegExp.test(record));
 
         // We currently only allow a single skylink to be tied to a domain name.
         // If there are more skylinks configured, raise an error.
-        if (dnslinks.length > 1) {
+        if (dnslinkSkylinks.length > 1) {
           return reject(
             new MultipleSkylinksError(
               `Multiple TXT records with valid skynet dnslink found for ${lookup}, only one allowed`
             )
           );
         }
 
-        const [dnslink] = dnslinks;
-        const matchSkylink = dnslink.match(dnslinkSkylinkRegExp);
-
-        // Verify if the configured skylink is valid.
-        if (!matchSkylink) {
-          return reject(
-            new InvalidSkylinkError(`TXT record with skynet dnslink for ${lookup} contains invalid skylink - ${hint}`)
-          );
-        }
-
-        const skylink = matchSkylink[1];
-
-        // Check if _dnslink records contain skynet-sponsor-key entries
-        const sponsors = records.filter((record) => sponsorRegExp.test(record));
-
         // We currently only allow a single sponsor key to be configured with a given domain name.
         // If there are more, raise an error.
-        if (sponsors.length > 1) {
+        if (dnslinkSponsors.length > 1) {
           return reject(
             new MultipleSponsorKeyRecordsError(
               `Multiple TXT records with valid sponsor key found for ${lookup}, only one allowed`
             )
           );
         }
 
-        if (sponsors.length === 1) {
-          // Extract just the key part from the record
-          const sponsor = sponsors[0].substring(sponsors[0].indexOf("=") + 1);
+        // If there are no dnslink records configured with our namespaces, we raise an error.
+        if (dnslinkSkylinks.length === 0 && dnslinkSponsors.length === 0) {
+          return reject(
+            new NoSkynetDNSLinksFoundError(
+              `TXT records for ${lookup} found but none of them contained valid skynet dnslink - ${hint}`
+            )
+          );
+        }
+
+        // Prepare response object
+        const response = {};
 
-          logger.info(`${domainName} => ${skylink} | sponsor: ${sponsor}`);
+        if (dnslinkSkylinks.length === 1) {
+          const [dnslinkSkylink] = dnslinkSkylinks;
+          const matchSkylink = dnslinkSkylink.match(dnslinkSkylinkRegExp);
```

Comment:
nit: is this more idiomatic in modern JS than
```js
const matchSkylink = dnslinkSkylinks[0].match(dnslinkSkylinkRegExp);
```
?
It looks so weird to me...

But in any case, it's not wrong! It's just an unexpected way to do this.

Reply:
it's the same I'd say, I personally don't like referencing array elements directly because I like to give them a specific name by assigning them to variable (being verbose about what is that element) but in this case it would have been the same